bin/update_dependencies.py

```diff
@@ -12,21 +14,39 @@
 
 # CUSTOM_COMPILE_COMMAND is a pip-compile option that tells users how to
 # regenerate the constraints files
-os.environ['CUSTOM_COMPILE_COMMAND'] = "bin/update_constraints.py"
-subprocess.check_call([
-    'pip-compile',
-    '--allow-unsafe',
-    '--upgrade',
-    'cibuildwheel/resources/constraints.in',
-])
-for python_version in ['27', '35', '36', '37']:
-    subprocess.check_call([
-        f'./env{python_version}/bin/pip-compile',
-        '--allow-unsafe',
-        '--upgrade',
-        'cibuildwheel/resources/constraints.in',
-        '--output-file', f'cibuildwheel/resources/constraints-python{python_version}.txt'
-    ])
+os.environ['CUSTOM_COMPILE_COMMAND'] = "bin/update_dependencies.py"
+
+PYTHON_VERSIONS = ['27', '35', '36', '37', '38', '39']
+
+if '--no-docker' in sys.argv:
+    for python_version in PYTHON_VERSIONS:
+        subprocess.check_call([
+            f'./env{python_version}/bin/pip-compile',
+            '--allow-unsafe',
+            '--upgrade',
+            'cibuildwheel/resources/constraints.in',
+            '--output-file', f'cibuildwheel/resources/constraints-python{python_version}.txt'
+        ])
+else:
+    image = 'quay.io/pypa/manylinux2010_x86_64:2020-12-19-8df9e2d'
```

Comment:
How do we update this? Do we update this, or is an older version OK?

Reply:
Any image version that has all python versions is ok. I only expect this one to require updating when adding a new python version so manually would do or maybe use `latest` because anyway, it will update the corresponding image to `latest` (well the properly tagged version of `latest`).

Reply:
Ah, yes. When adding a Python version, we'll have to update the Python versions in `update_dependencies.py` as well, anyway, now.

Reply:
`update_update_dependencies.py`? ;-)

Reply:
Can’t this be latest? This is a case where pinning absolutely everything might be too much. :)

Reply:
done